src/Datadog.Trace.ClrProfiler.Native/metadata_builder.cpp

```diff
@@ -122,25 +122,26 @@ HRESULT MetadataBuilder::StoreWrapperMethodRef(
 
   member_ref = mdMemberRefNil;
 
-  hr = metadata_import_->FindMemberRef(
-      type_ref, method_replacement.wrapper_method.method_name.c_str(),
-      method_replacement.wrapper_method.method_signature.data.data(),
-      (DWORD)(method_replacement.wrapper_method.method_signature.data.size()),
-      &member_ref);
-
-  if (hr == HRESULT(0x80131130) /* record not found on lookup */) {
-    // if memberRef not found, create it by emitting a metadata token
-    hr = metadata_emit_->DefineMemberRef(
-        type_ref, method_replacement.wrapper_method.method_name.c_str(),
-        method_replacement.wrapper_method.method_signature.data.data(),
-        (DWORD)(method_replacement.wrapper_method.method_signature.data.size()),
-        &member_ref);
-  }
-
-  if (FAILED(hr)) {
-    // Record that this cache_key failed
-    metadata_.SetFailedWrapperMemberKey(cache_key);
-    return hr;
+  auto signature_data = method_replacement.wrapper_method.method_signature.data;
+  if (signature_data.size() > 0) {
+      hr = metadata_import_->FindMemberRef(
+          type_ref, method_replacement.wrapper_method.method_name.c_str(),
+          signature_data.data(), (DWORD)(signature_data.size()),
+          &member_ref);
+
+      if (hr == HRESULT(0x80131130) /* record not found on lookup */) {
+        // if memberRef not found, create it by emitting a metadata token
+        hr = metadata_emit_->DefineMemberRef(
+            type_ref, method_replacement.wrapper_method.method_name.c_str(),
+            signature_data.data(), (DWORD)(signature_data.size()),
+            &member_ref);
+      }
+
+      if (FAILED(hr)) {
+        // Record that this cache_key failed
+        metadata_.SetFailedWrapperMemberKey(cache_key);
+        return hr;
+      }
```

Comment:
This should be reverted in favor of this PR: https://github.com/DataDog/dd-trace-dotnet/pull/1259

Reply:
Done